packages/-ember-data/tests/integration/references/record-test.js

```diff
@@ -32,75 +33,134 @@ module('integration/references/record', function (hooks) {
     assert.equal(recordReference.id(), 1);
   });
 
-  test('push(object)', function (assert) {
-    var done = assert.async();
+  test('a RecordReference can be retrieved via store.getReference(identifier) without local state', function (assert) {
+    let store = this.owner.lookup('service:store');
+    let recordReference = store.getReference({ type: 'person', id: '1' });
+
+    assert.equal(recordReference.remoteType(), 'identity');
+    assert.equal(recordReference.type, 'person');
+    assert.equal(recordReference.id(), '1');
+  });
+
+  [
+    { withType: true, withId: true, withLid: true, desc: 'type, id and lid' },
+    { withType: true, withLid: true, desc: 'type and lid' },
+    { withType: true, withLid: true, isCreate: true, desc: 'type and lid via store.createRecord (no local id)' },
+    {
+      withType: true,
+      withLid: true,
+      isCreate: true,
+      generateLid: true,
+      desc: 'type and generated lid via store.createRecord (no local id)',
+    },
+    { withType: true, withLid: true, generateLid: true, desc: 'type and generated lid' },
+    { withType: true, withId: true, withLid: true, desc: 'type, id and lid from cache' },
+    { withType: true, withLid: true, exta: { id: null }, desc: 'type, null id, and lid' },
+    {
+      withType: true,
+      withLid: true,
+      exta: { id: null },
+      isCreate: true,
+      desc: 'type, null id, and lid via store.createRecord',
+    },
+  ].forEach(({ withType, withId, withLid, extra, isCreate, generateLid, desc }) => {
```

Comment:
extra will always be undefined as above it is set as `exta`. We should just remove it entirely as it is functioning identically to the idea of `withId` and where we've used it it has been the same as `withId: undefined` which is falsey. No id present in the args used to build the resource-identifier vs null-id isn't material here, and the tests currently are using the no-id present path anyway :)